repository/src/main/java/org/apache/atlas/repository/util/AccessControlUtils.java

```diff
@@ -0,0 +1,406 @@
+    }
+
+    public static List<AtlasEntity> objectToEntityList(AtlasEntity.AtlasEntityWithExtInfo entityWithExtInfo, List<AtlasObjectId> policies) {
+        List<AtlasEntity> ret = new ArrayList<>();
+
+        Set<String> referredGuids =  entityWithExtInfo.getReferredEntities().keySet();
+        if (policies != null) {
+            ret = policies.stream()
+                    .filter(x -> referredGuids.contains(x.getGuid()))
+                    .map(x -> entityWithExtInfo.getReferredEntity(x.getGuid()))
+                    .filter(x -> x.getStatus() == null || x.getStatus() == AtlasEntity.Status.ACTIVE)
+                    .collect(Collectors.toList());
+        }
+
+        return ret;
+    }
+
+    public static List<String> getPurposeTags(AtlasStruct entity) {
+        return getListAttribute(entity, ATTR_PURPOSE_CLASSIFICATIONS);
+    }
+
+    public static boolean getIsAccessControlEnabled(AtlasEntity entity) {
+        return (boolean) entity.getAttribute(ATTR_ACCESS_CONTROL_ENABLED);
+    }
+
+    public static List<String> getPersonaUsers(AtlasStruct entity) {
+        return getListAttribute(entity, ATTR_PERSONA_USERS);
+    }
+
+    public static List<String> getPersonaGroups(AtlasStruct entity) {
+        return getListAttribute(entity, ATTR_PERSONA_GROUPS);
+    }
+
+    public static String getPersonaRoleId(AtlasEntity entity) {
+        String roleId = (String) entity.getAttribute(ATTR_PERSONA_ROLE_ID);
+        if (roleId == null) {
+            LOG.warn("roleId not found for Persona with GUID " + entity.getGuid());
+        }
+        return roleId;
+    }
+
+    public static String getTenantId(AtlasStruct entity) {
+        String ret = DEFAULT_TENANT_ID;
+
+        Object tenantId = entity.getAttribute(ATTR_TENANT_ID);
+
+        if (tenantId != null) {
+            String tenantIdAsString = (String) tenantId;
+            if (tenantIdAsString.length() > 0) {
+                ret = tenantIdAsString;
+            }
+        }
+
+        return ret;
+    }
+
+    public static void validateNoPoliciesAttached(AtlasEntity entity) throws AtlasBaseException {
+        List<AtlasObjectId> policies = (List<AtlasObjectId>) entity.getRelationshipAttribute(REL_ATTR_POLICIES);
+        if (CollectionUtils.isNotEmpty(policies)) {
+            throw new AtlasBaseException(OPERATION_NOT_SUPPORTED, "Can not attach a policy while creating/updating Persona/Purpose");
+        }
+    }
+
+    public static AtlasEntity getConnectionForPolicy(EntityGraphRetriever entityRetriever, List<String> resources) throws AtlasBaseException {
+        AtlasEntity ret = null;
+        if (CollectionUtils.isNotEmpty(resources)) {
+
+            String entityId = resources.get(0).split(RESOURCES_SPLITTER)[1];
+
+            ret = extractConnectionFromResource(entityRetriever, entityId);
+        }
+
+        return ret;
+    }
+
+    public static String getUUID(){
+        return NanoIdUtils.randomNanoId(22);
```

Comment:
why 22 ?

Reply:
This was copied from an external library - https://github.com/aventrix/jnanoid/blob/develop/src/main/java/com/aventrix/jnanoid/jnanoid/NanoIdUtils.java
We already use 21 size for Classifications, BMs
So just an extra step to be safe in terms of uniqueness we are using 22 for Persona/Purpose

Reply:
21 was default from original code